lte/gateway/python/magma/mobilityd/ipv6_allocator_pool.py

```diff
@@ -0,0 +1,191 @@
+        self._ipv6_session_prefix_alloc_mode = self._config[
+            'ipv6_ip_allocator_type']
+
+    def add_ip_block(self, ipblock: ip_network):
+        """
+        Adds IP block to the assigned IP block of the IPv6 allocator
+        :param ipblock: IPv6 ip_network object
+        """
+        if self._assigned_ip_block and ipblock.overlaps(
+                self._assigned_ip_block):
+            logging.error("Overlapped IP block: %s", ipblock)
+            raise OverlappedIPBlocksError(ipblock)
+
+        if ipblock.prefixlen > MAX_IPV6_CONF_PREFIX_LEN:
+            msg = "IPv6 block exceeds maximum allowed prefix length"
+            logging.error(msg)
+            raise InvalidIPv6NetworkError(msg)
+
+        # For now only one IPv6 network is supported
+        self._assigned_ip_block = ipblock
+
+    def remove_ip_blocks(self, *ipblocks: List[ip_network],
+                         force: bool = False) -> List[ip_network]:
+        """
+        Removes assigned IP block (as it only supports one for now)
+        :param ipblocks: List of ip_network objects to remove
+        :param force:
+        :return: Removed list of IP blocks
+        """
+        removed_blocks = []
+        if self._assigned_ip_block in ipblocks:
+            # Clear allocated session prefix and IID store
+            self._allocated_iid.clear()
+            self._sid_session_prefix_allocated.clear()
+
+            removed_blocks.append(self._assigned_ip_block)
+            self._assigned_ip_block = None
+        return removed_blocks
+
+    def alloc_ip_address(self, sid: str, _) -> IPDesc:
+        """
+        Calculates full IPv6 Address from configured prefix part
+        (assigned_ip_block) + unique session prefix part + interface
+        identifier
+
+        :param sid: composite SID (IMSI + apn)
+        :param _:
+        :return: Full IPv6 address
+        """
+        # Take available ipv6 host from network
+        ipv6_addr_part = next(self._assigned_ip_block.hosts())
+
+        # Calculate session part from rest of 64 prefix bits
+        session_prefix_part = self._get_session_prefix(sid)
+        if not session_prefix_part:
+            logging.error('Could not get IPv6 session prefix for sid: %s', sid)
+            raise MaxCalculationError(
+                'Could not get IPv6 session prefix for sid: %s', sid)
+
+        # Get interface identifier from 64 bits fixed length
+        iid_part = self._get_ipv6_iid_part(IID_PART_LEN)
+        if not iid_part:
+            logging.error('Could not get IPv6 IID for sid: %s', sid)
+            raise MaxCalculationError(
+                'Could not get IPv6 IID for sid: %s', sid)
+
+        ipv6_addr = ipv6_addr_part + (session_prefix_part * iid_part)
+        ip_desc = IPDesc()
+        ip_desc.sid = sid
+        ip_desc.ip = ipv6_addr
+        ip_desc.ip_block = self._assigned_ip_block
+        ip_desc.type = IPType.IP_POOL
+        return ip_desc
+
+    def release_ip(self, ip_desc: IPDesc):
```

Comment:
we need to hold released IP same as IPv4, reasoning is here: https://github.com/magma/magma/blob/8b14ea3f18391df3f993a5308cc9587057ba2e8e/lte/gateway/python/magma/mobilityd/ip_address_man.py#L15

Reply:
I'm not sure how that would apply for IPv6 as we're not holding the full IPv6 address but rather tracking the allocated IID and session prefix parts, plus tracking the full IPv6 addresses doesn't seem efficient as there can be a lot. Let's discuss offline